repeat_expansion_variants/pipeline.py

```diff
@@ -0,0 +1,209 @@
+#!/usr/bin/env python3
+"""A pipeline to extract repeat expansion variants from ClinVar TSV dump. For documentation refer to README.md"""
+
+import gzip
+from io import StringIO
+import logging
+
+import numpy as np
+import pandas as pd
+
+from . import biomart, clinvar_identifier_parsing
+
+logging.basicConfig()
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.INFO)
+
+
+def none_to_nan(*args):
+    """Converts all arguments which are None to np.nan, for consistency inside a Pandas dataframe."""
+    return [np.nan if a is None else a for a in args]
+
+
+def load_clinvar_data(clinvar_summary_tsv):
+    """Load ClinVar data, preprocess, and return it as a Pandas dataframe."""
+    # Load and pre-filter the file
+    gzip_stream = gzip.open(clinvar_summary_tsv, 'r')
+    repeat_expansion_data = ''.join([
+        line.decode() for i, line in enumerate(gzip_stream)
+        if i == 0 or line.decode().split('\t')[1] == 'NT expansion'
+    ])
+    # Load variants
+    variants = pd.read_table(StringIO(repeat_expansion_data), low_memory=False)
+    # Filter only NT expansion variants
+    variants = variants[variants['Type'] == 'NT expansion']
+    # Drop all columns except the ones we require
+    variants = variants[['Name', 'RCVaccession', 'GeneSymbol', 'HGNC_ID']]
+    # Records may contain multiple RCVs per row, delimited by semicolon. Here we explode them into separate rows
+    variants['RCVaccession'] = variants['RCVaccession'].str.split(';')
```

Comment:
I suggest removing empty entries (if that is possible) resulting from splitting a string like "RCVaccession1:RCVaccession2:"

Reply:
As far as I can see, ClinVar files never contain such a case, it's always something like `RCV000144122;RCV000169953`, without a trailing semicolon. I don't suppose we should add support for data peculiarities which we don't observe.

However, just to be extra _extra_ safe, I've added an assertion prior to outputting the final consequences table that none of the values in it are empty: https://github.com/EBIvariation/vep-mapping-pipeline/pull/3/commits/45e54e50507450b2530ee10145ca04e5d01d9426. This is a useful assert to have anyway.